src/Microsoft.Azure.ServiceBus/Core/MessageReceiver.cs

```diff
@@ -1455,19 +1453,15 @@ Rejected GetRejectedOutcome(IDictionary<string, object> propertiesToModify, stri
                     rejected.Error.Info.Add(Message.DeadLetterErrorDescriptionHeader, deadLetterErrorDescription);
                 }
 
-                if (propertiesToModify != null)
+                foreach (var pair in propertiesToModify)
```

Comment:
This has to be wrapped inside `if (propertiesToModify != null)`

Reply:
Silly mistake, sorry. Fixing.